### PR TITLE
MM30518 Adding preferences tab to playbook edit.

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary
Adds preferences tab to playbook edit. Preferences page is just a placeholder. 
Also changed the titles in the "Permissions" sidebar to their new names. 

Screenshot:
image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30518

### CODE REVIEW

File: webapp/src/components/tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+export interface TabsContentProps {
+    children: React.ReactNodeArray;
+    currentTab: number;
+}
+
+export const TabsContent: FC<TabsContentProps> = (props: TabsContentProps) => {
+    return (
+        <>
+            {props.children[props.currentTab]}
+        </>
```
Comment (cpoile) [inline webapp/src/components/tabs.tsx:67]: Do we need the fragment here? It should know that `props.children[props.currentTab]` will return a ReactNode, right?

File: webapp/src/components/backstage/playbook_edit.tsx [hunk 2]
```diff
@@ -386,30 +389,37 @@ const PlaybookEdit: FC<Props> = (props: Props) => {
...
-                            onChange={updateChecklist}
-                        />
+                        <TabsContent
+                            currentTab={currentTab}
+                        >
+                            <StagesAndStepsEdit
+                                checklists={playbook.checklists}
+                                onChange={updateChecklist}
+                            />
+                            {'TODO MM-30519 implment perferences.'}
```
Comment (cpoile) [inline webapp/src/components/backstage/playbook_edit.tsx:409]:
```suggestion
                            {'TODO MM-30519 implement perferences.'}
```
😛

**Review by cpoile — APPROVED** [review]
Wow, that's amazingly concise. I love it.

File: webapp/src/components/tabs.tsx (hunk 1 above)
Comment (crspeller) [reply webapp/src/components/tabs.tsx:67]: For some reason typescript wants it.

**Review by agarciamontoro — APPROVED** [review]
Nice and clean! :tada:

File: webapp/src/components/backstage/playbook_edit.tsx [hunk 3]
```diff
@@ -386,30 +389,37 @@ const PlaybookEdit: FC<Props> = (props: Props) => {
...
-                            <EditHeaderHelpText>{'Stages allow you to group your tasks. Tasks are meant to be completed by members of the incident channel.'}</EditHeaderHelpText>
-                        </EditHeaderTextContainer>
-                    </EditHeader>
+                    <TabsHeader>
+                        <Tabs
+                            currentTab={currentTab}
+                            setCurrentTab={setCurrentTab}
+                        >
+                            {'Tasks'}
+                            {'Perferences'}
```
Comment (agarciamontoro) [inline webapp/src/components/backstage/playbook_edit.tsx:398]:
```suggestion
                            {'Preferences'}
```

File: webapp/src/components/backstage/playbook_edit.tsx [hunk 4]
```diff
@@ -386,30 +389,37 @@ const PlaybookEdit: FC<Props> = (props: Props) => {
...
-                            onChange={updateChecklist}
-                        />
+                        <TabsContent
+                            currentTab={currentTab}
+                        >
+                            <StagesAndStepsEdit
+                                checklists={playbook.checklists}
+                                onChange={updateChecklist}
+                            />
+                            {'TODO MM-30519 implement perferences.'}
```
Comment (agarciamontoro) [inline webapp/src/components/backstage/playbook_edit.tsx:409]: This is gone soon, but:
```suggestion
                            {'TODO MM-30519 implement preferences.'}
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Oopsie, just saw a couple of typos!

**Review by agarciamontoro — APPROVED** [review]
:tada:

**Review by itao — APPROVED** [review]
Looks great!

**Pull request merged**